Fix docstring errors in container.py, _functions.py, transformer.py, comm.py, parallel_apply.py, data_parallel.py, scatter_gather.py

markstur opened the pull request:
Fix docstring errors in container.py, _functions.py, transformer.py, comm.py, parallel_apply.py, data_parallel.py, scatter_gather.py

Fixes #112603



cc @svekars @carljparker

markstur committed:
Fix docstring errors in container.py, _functions.py, transformer.py, comm.py, parallel_apply.py, data_parallel.py, scatter_gather.py

Fixes #112603

Signed-off-by: markstur <mark.sturdevant@ibm.com>

markstur commented:
@pytorchbot label "release notes: nn"

svekars commented on torch/nn/modules/transformer.py:
```diff
@@ -215,22 +214,18 @@ def generate_square_subsequent_mask(
 class TransformerEncoder(Module):
-    r"""TransformerEncoder is a stack of N encoder layers. Users can build the
-    BERT(https://arxiv.org/abs/1810.04805) model with corresponding parameters.
+    r"""TransformerEncoder is a stack of N encoder layers. Users can build the BERT(https://arxiv.org/abs/1810.04805) model with corresponding parameters.
```
```suggestion
    r"""TransformerEncoder is a stack of N encoder layers. 
    
    Users can build the BERT(https://arxiv.org/abs/1810.04805) model with corresponding parameters.
```

svekars commented on torch/nn/modules/transformer.py:
```diff
@@ -215,22 +214,18 @@ def generate_square_subsequent_mask(
-        r"""Generate a square causal mask for the sequence. The masked positions are filled with float('-inf').
-            Unmasked positions are filled with float(0.0).
-        """
+        r"""Generate a square causal mask for the sequence. The masked positions are filled with float('-inf'). Unmasked positions are filled with float(0.0)."""
```
```suggestion
        r"""Generate a square causal mask for the sequence. 
        
        The masked positions are filled with float('-inf'). Unmasked positions are filled with float(0.0)."""
```

svekars commented on torch/nn/parallel/comm.py:
```diff
@@ -38,9 +38,7 @@ def broadcast(tensor, devices=None, *, out=None):
-    """Broadcasts a sequence tensors to the specified GPUs.
-    Small tensors are first coalesced into a buffer to reduce the number
-    of synchronizations.
+    """Broadcasts a sequence tensors to the specified GPUs. Small tensors are first coalesced into a buffer to reduce the number of synchronizations.
```
```suggestion
    """Broadcasts a sequence tensors to the specified GPUs. 
    
    Small tensors are first coalesced into a buffer to reduce the number of synchronizations.
```

svekars commented on torch/nn/parallel/scatter_gather.py:
```diff
@@ -33,11 +33,7 @@ def scatter(inputs: T, target_gpus: Sequence[Union[int, torch.device]], dim: int
-    distributes them across given GPUs. Duplicates
-    references to objects that are not tensors.
-    """
+    r"""Slices tensors into approximately equal chunks and distributes them across given GPUs. Duplicates references to objects that are not tensors."""
```
```suggestion
    r"""Slices tensors into approximately equal chunks and distributes them across given GPUs. 
    
    Duplicates references to objects that are not tensors."""
```

markstur replied:
also s/Slices/Slice/

svekars commented on torch/nn/parallel/scatter_gather.py:
```diff
@@ -80,10 +76,7 @@ def scatter_kwargs(
-    Gathers tensors from different GPUs on a specified device.
-    Use 'cpu' for CPU to avoid a deprecation warning.
-    """
+    r"""Gather tensors from different GPUs on a specified device. Use 'cpu' for CPU to avoid a deprecation warning."""
```
```suggestion
    r"""Gather tensors from different GPUs on a specified device. 
    
    Use 'cpu' for CPU to avoid a deprecation warning."""
```

svekars reviewed (commented):
A couple of suggestions. Thanks!

markstur committed:
Fix docstrings (shorter summary, separate details)

Docstring tweaks.
* Mostly multi-sentence split in to summary and detail.
* Some imperative mode changes.

Signed-off-by: markstur <mark.sturdevant@ibm.com>

markstur commented:
> Reference

Thanks!  Updated and also commented on some imperative mood things that I fixed (and one missing "of").  Hopefully those are improvements too.

kit1980 commented:
@markstur Please fix the "line too long" lint issues.

markstur committed:
Fix line-too-long problems.

Signed-off-by: markstur <mark.sturdevant@ibm.com>

mikaylagawarecki commented on torch/nn/modules/container.py:
```diff
@@ -836,23 +828,19 @@ def fromkeys(self, keys: Iterable[str], default: Optional[Any] = None) -> 'Param
     def update(self, parameters: Union[Mapping[str, Any], 'ParameterDict']) -> None:
-        r"""Update the :class:`~torch.nn.ParameterDict` with the key-value pairs from a
-        mapping or an iterable, overwriting existing keys.
+        r"""Update the :class:`~torch.nn.ParameterDict` with key-value pairs from parameters, overwriting existing keys.
```
```suggestion
        r"""Update the :class:`~torch.nn.ParameterDict` with key-value pairs from ``parameters``, overwriting existing keys.
```

mikaylagawarecki reviewed (approved):
LGTM

svekars committed:
Update torch/nn/modules/container.py

Co-authored-by: mikaylagawarecki <mikaylagawarecki@gmail.com>

svekars commented:
@pytorchbot merge
